msl/equipment/resources/isotech/millik.py

```diff
@@ -147,7 +147,7 @@ def read_all_channels(self, n: int = 1) -> list[float]:
             else:       # average multiple readings
                 results.append(sum(readings)/len(readings))
 
-        return results
+        return sorted(self.channel_configuration), results
```

Comment:
It may be safer to create a local variable, e.g, 
```python
channels=sorted(self.channel_configuration)

for c in channels:
   # your code...

return channels, results
```
that way there cannot be any potential issues if someone else (in the future) modifies the _for_ loop and they forget that the return object is expected to have the same channel order.

Reply:
Good thinking. I think I subconsciously wondered about that as I copied the sorted() bit to the return statement, but was preoccupied with whether to return a tuple of lists or a list of tuples!